rest_docs/rest-catalog-open-api.yaml

```diff
@@ -367,6 +367,90 @@ paths:
 
+    get:
+      tags:
+        - Catalog API
+      summary: Load a table from the catalog
+      operationId: loadTable
+      description:
+        Load a table from the catalog.
+
+        The response contains both configuration and table metadata. The configuration, if non-empty is used
+        as additional configuration for the table that overrides catalog configuration. For example, this
+        configuration may change the FileIO implemented used for the table.
+
+        The response also contains the table's full metadata.
+      responses:
+        200:
+          $ref: '#/components/responses/LoadTableResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        404:
+          description:
+            Not Found - NoSuchTableException, table to load does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergErrorResponse'
+              examples:
+                TableToRenameDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+        5XX:
+          $ref: '#/components/responses/ServerErrorResponse'
+
+    post:
+      tags:
+        - Catalog API
+      summary: Commit updates to a table
+      operationId: updateTable
+      description:
+        Commit updates to a table.
+
+        Commits have two parts, requirements and updates. Requirements are assertions that will be validated
+        before attempting to make and commit changes. For example, assert-ref-snapshot-id will check that a
+        named ref's snapshot ID has a certain value.
+
+        Updates are changes to make to table metadata. For example, after asserting that the current main ref
+        is at the expected snapshot, a commit may add a new child snapshot and set the ref to the new
+        snapshot id.
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CommitTableRequest'
+      responses:
+        200:
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/CommitTableResponse'
+        400:
+          $ref: '#/components/responses/BadRequestErrorResponse'
+        401:
+          $ref: '#/components/responses/UnauthorizedResponse'
+        404:
+          description:
+            Not Found - NoSuchTableException, table to load does not exist
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/responses/IcebergErrorResponse'
+              examples:
+                TableToRenameDoesNotExist:
+                  $ref: '#/components/examples/NoSuchTableError'
+        409:
+          description:
+            Conflict - CommitFailedException, one or more requirements failed. The client may retry.
```

Comment:
Good question. I think that we do have the possibility of unknown commit state for the same reasons we have it in the other catalog clients. If this is sending the update off to a DB and gets disconnected, then there's a possibility that the commit succeeded but the service doesn't know about it. We can't guarantee that the service endpoint always knows the final commit state.

However, I don't think that this should actually send `CommitStateUnknown`. Instead, it should send a 5xx error and trigger commit state recovery on the client side. That way, we don't have retries within retries. The client would get the complete server-side error, then query the table to see whether the commit succeeded. If it did, then it would return success and if not would throw its own `CommitStateUnknownException`.

Reply:
So we should probably put a note in 50X Server Error, Operation may or may not have succeeded? Recheck before retrying? I just feel like we need to specify it so implementers know.

Reply:
In the previous PR, we put `50X` everywhere. I believe I also included a section that mentions 50X can return text/plain (eg some middleware or proxy server hit an issue and then returned an HTML page, which definitely happens even once you hit the app sever).

I think we should include them everywhere, but I can understand possibly adding them last as they’re the same throughout,

50X response would be maybe a good argument for using an external tool to bundle all of the files together, though we’re likely far from being in that position currently and should focus on defining the spec first.

Reply:
> Instead, it should send a 5xx error and trigger commit state recovery on the client side.

Yes a 5XX sounds like the right way for commit state unknown. This is probably a good one to distinguish 503 and 500, where 500 represents the service is likely down and commit state is unknown, 503 means the user is throttled maybe because of too many requests, but we are sure the commit failed.

Reply:
with that being said, it seems like we should always distinguish 500 and 503? Just realized what I said is probably true for all the other APIs, but this one we want to return a different Java exception to users based on the code.

Reply:
I updated this so that the description documents what to do. After looking through the list of [status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes), I think that only 500 (generic error) and 504 (gateway timeout) should leave the table in an unknown state. Other error codes indicate that the operation wasn't attempted, like 501 (not implemented).